uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/manager-leave/components/manager-leave-requests_test.js

```diff
@@ -1,30 +1,899 @@
+        spyOn(AbsenceType, 'all').and.callThrough();
+
+        spyOn(Contact, 'all').and.callFake(function () {
+          return $q.resolve(ContactAPIMock.mockedContacts());
+        });
+
+        spyOn(OptionGroup, 'valuesOf').and.callFake(function (name) {
+          return OptionGroupAPIMock.valuesOf(name);
+        });
+      });
+
+      beforeEach(function () {
         compileComponent();
-      }));
+      });
 
       it('is initialized', function () {
         expect($log.debug).toHaveBeenCalled();
       });
 
+      describe('initially', function () {
+
+        it('the filter section is closed', function () {
+          expect(controller.isFilterExpanded).toBe(false);
+        });
+
+        describe('pagination', function () {
+
+          it('is set to page 1', function () {
+            expect(controller.pagination.page).toBe(1);
+          });
+
+          it('has page size set to 7', function () {
+            expect(controller.pagination.size).toBe(7);
+          });
+        });
+      });
+
+      describe('data loading::', function () {
+
+        //TODO need to figure out how to test variables which are changed when controller gets initialized
+        xdescribe('before loading starts', function () {
+
+          it('loading should be hidden', function () {
+
+          });
+
+          it('leave requests are empty', function () {
+
+          });
+
+          it('absencePeriods are empty', function () {
+
+          });
+
+          it('absenceTypes are empty', function () {
+
+          });
+
+          it('statusCount is reset', function () {
+
+          });
+        });
+
+        describe('after data loading is complete', function () {
+
+          describe('loading', function () {
+
+            it('should be hidden for page', function () {
+              expect(controller.loading.page).toBe(false);
+            });
+
+            it('should be hidden for content', function () {
+              expect(controller.loading.content).toBe(false);
+            });
+          });
+
+          it('leave requests status should load', function () {
+            var expectedResult = optionGroupMock.getCollection('hrleaveandabsences_leave_request_status').concat({
+              name: 'all',
+              label: 'All'
+            });
+
+            expect(controller.leaveRequestStatuses).toEqual(expectedResult);
+          });
+
+          it('regions should load', function () {
+            expect(controller.regions).toEqual(optionGroupMock.getCollection('hrjc_region'));
+          });
+
+          it('departments should load', function () {
+            expect(controller.departments).toEqual(optionGroupMock.getCollection('hrjc_department'));
+          });
+
+          it('locations should load', function () {
+            expect(controller.locations).toEqual(optionGroupMock.getCollection('hrjc_location'));
+          });
+
+          it('levelTypes should load', function () {
```

Comment:
do not use variables in the test description if it's not absolutely necessary (as in, you are specifically testing some variable). Always write the test titles in plain english
```js
it('has loaded the job role levels', function () {});
it('has loaded the absence periods', function () {});
it('has loaded the absence types', function () {});